integrations/va-auth0/src/index.tsx

```diff
@@ -213,78 +252,116 @@ const handleFetchEvent: FetchEventCallback<Auth0RuntimeContext> = async (request
-                                status: 401,
-                            });
-                        });
+                }
 
-                    if ('access_token' in resp) {
-                        let url;
-                        if (request.query.state) {
-                            url = new URL(
-                                `${publishedContentUrls?.published}${request.query.state}`,
-                            );
-                            url.searchParams.append('jwt_token', token);
-                        } else {
-                            url = new URL(publishedContentUrls?.published!);
-                            url.searchParams.append('jwt_token', token);
-                        }
-                        if (publishedContentUrls?.published && token) {
-                            return Response.redirect(url.toString());
-                        } else {
-                            return new Response(
-                                "Error: Either JWT token or space's published URL is missing",
-                                {
-                                    status: 500,
-                                },
-                            );
-                        }
-                    } else {
-                        logger.debug(JSON.stringify(resp, null, 2));
-                        logger.debug(
-                            `Did not receive access token. Error: ${(resp && resp.error) || ''} ${
-                                (resp && resp.error_description) || ''
-                            }`,
-                        );
+                let userInfo;
+                if (includeClaimsInToken) {
+                    // Auth0 returns an opaque access token (i.e., one that doesn't include user or custom claims) when
+                    // exchanging an authorization code for a token—unless an audience (aud parameter) is set to a valid
+                    // Auth0 API application during the authorization request.
+                    // In this case, since we only request a token to verify authentication and generate the VA,
+                    // there is no valid API target to specify as audience.
+                    // As a result, we must call the /userinfo endpoint (an OIDC-compliant endpoint) to retrieve user claims.
+                    //
+                    // An alternative approach would be to request an ID Token using the Implicit Grant flow (with form post).
+                    // However, many authentication providers discourage this method in favor of the Authorization Code flow.
+                    // Additionally, some customers may disable the Implicit Grant flow in their Auth0 application.
+                    // Therefore, retrieving user data via the /userinfo endpoint is a more robust solution.
+                    const auth0TokenData = await auth0TokenResp.json<Auth0TokenResponseData>();
```

Comment:
Curious why we don't use a auth0 library or something

Reply:
The previous code didn't use a library hence why I didn't look into using one. I haven't honestly looked into why we didn't use one before but it could have been aimed at reducing the integration bundle size or just that we didn't see the need to since we are only relying on a couple of endpoints, or maybe that none were found to be compatible with the Cloudflare worker runtime. Also I had very quick look at the Auth0 libraries and they seem to be focus on providing a middleware only but didn't look in details.

I'll look at refactoring the integration code to use a library in another PR.

Reply:
@SamyPesse @spastorelli We don't use an auth0 library because Cloudflare works runtime only supports a few libraries, and Auth0 isn't one of them. It is also why for the JWT library we're using a lib that's built specifically for CF